STEER/STEER/AliMC.cxx

```diff
@@ -408,8 +408,227 @@ void  AliMC::AddParticles()
   TVirtualMC::GetMC()->SetDecayMode(-9322134,psratio,psmode);
   TVirtualMC::GetMC()->SetDecayMode(-9322136,psratio,psmode);
 
+	
+  //Ps+- and Ps0 Hidden Strange Pentaquarks
+  for (int j=0; j<7; j++) {
+  psmode[j][0] = psmode[j][1] = 0;
+  psratio[j] = 0.;
+      }
+
+  TVirtualMC::GetMC()->DefineParticle( 9221332, "Ps_1670", kPTHadron, 1.67 ,  1.0, 6.582e-24, "Hadron", 0.1, 1, -1, 0, 1, -1, 0, 0,  1, kTRUE);
+  psmode[0][0] = 3122; // Lambda
+  psmode[0][1] = 321; // K+
+  psratio[0] = 26.85695751;
+  psmode[1][0] = 3312; // Xi-
+  psmode[1][1] = -211; // Pi-
+  psratio[1] = 73.14304249;
+  TVirtualMC::GetMC()->SetDecayMode(9221332,psratio,psmode);
+    
+    
+  TVirtualMC::GetMC()->DefineParticle( 92213320, "Ps_0_1670", kPTNeutron, 1.67 ,  0.0, 6.582e-24, "Hadron", 0.1, 1, -1, 0, 1, -1, 0, 0,  1, kTRUE);
+  psmode[0][0] = 3122; // Lambda
+  psmode[0][1] = 310; // K0S
+  psratio[0] = 49.01650463;
+  psmode[1][0] = 3312; // Xi-
+  psmode[1][1] = 211; // Pi+
+  psratio[1] = 50.98349537;
+  TVirtualMC::GetMC()->SetDecayMode(92213320,psratio,psmode);
+    
+    
+
+  TVirtualMC::GetMC()->DefineParticle( 9223132, "Ps_1870", kPTHadron, 1.87 ,  1.0, 6.582e-24, "Hadron", 0.1, 1, -1, 0, 1, -1, 0, 0,  1, kTRUE);
+  psmode[0][0] = 3122; // Lambda
+  psmode[0][1] = 321; // K+
+  psratio[0] = 2.27560051;
+  psmode[1][0] = 3312; // Xi-
+  psmode[1][1] = -211; // Pi-
+  psratio[1] = 6.19743858;
+  psmode[2][0] = 3224; // Sigma*+
+  psmode[2][1] = 310; // K0
+  psratio[2] =  46.98510416;
+  psmode[3][0] = 3114; // Sigma*-
+  psmode[3][1] = 310; // K0
+  psratio[3] =  44.54185676;
+  TVirtualMC::GetMC()->SetDecayMode(9223132,psratio,psmode);
+    
+    
+  TVirtualMC::GetMC()->DefineParticle( 92231320, "Ps_0_1870", kPTNeutron, 1.87 ,  0.0, 6.582e-24, "Hadron", 0.1, 1, -1, 0, 1, -1, 0, 0,  1, kTRUE);
```

Comment:
Before here, don't you need to re-initialize the psmode and psratio? At least you changed above the 3-rd element, which will then go also in the decay mode below.

Reply:
Yes, I see what you are saying, good catch. Perhaps instead of reinitializing (which will work), I could rearrange from fewest to most decay channels? This will make for fewer lines and still passing wrong decay modes.

Reply:
Hello @JakeMartinez,
I think that the cleanest is to reinitialize. 
Chiara